src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.cs

```diff
@@ -264,22 +264,54 @@ private void ProcessEvent(WorkspaceChangeEventArgs args, string eventName)
                 switch (args.Kind)
                 {
                     case WorkspaceChangeKind.SolutionAdded:
+                        EnqueueFullSolutionEvent(args.NewSolution, InvocationReasons.DocumentAdded, eventName);
+                        break;
+
                     case WorkspaceChangeKind.SolutionChanged:
                     case WorkspaceChangeKind.SolutionReloaded:
+                        EnqueueSolutionChangedEvent(args.OldSolution, args.NewSolution, eventName);
+                        break;
+
                     case WorkspaceChangeKind.SolutionRemoved:
+                        EnqueueFullSolutionEvent(args.OldSolution, InvocationReasons.SolutionRemoved, eventName);
+                        break;
+
                     case WorkspaceChangeKind.SolutionCleared:
-                        ProcessSolutionEvent(args, eventName);
+                        EnqueueFullSolutionEvent(args.OldSolution, InvocationReasons.DocumentRemoved, eventName);
```

Comment:
Ditto to earlier "surprise" that solution add/remove is using the Document InvocationReasons.

Is there any meaningful difference here between SolutionRemoved and SolutionCleared? I'm not sure I know what the difference between those are myself.